File: ALittle/Module/Core/Script/ALittle.lua
-- ALittle Generate Lua
local ___pairs = pairs
local ___ipairs = ipairs


function RequireCore(base_path)
local ___COROUTINE = coroutine.running()
	ALittle.Require(base_path, "Core/Reflect/ReflectRegister")
	ALittle.Require(base_path, "Core/Reflect/ReflectDefine")
	ALittle.Require(base_path, "Core/Lua/LuaBind")
	ALittle.Require(base_path, "Core/Lua/LuaClass")
	ALittle.Require(base_path, "Core/Lua/LuaException")
	if _G["bit"] == nil then
		_G["bit"] = _G["bit32"]
	end
	ALittle.Require(base_path, "Core/Utility/Log")
	ALittle.Require(base_path, "Core/Utility/List")
	ALittle.Require(base_path, "Core/Utility/Map")
	ALittle.Require(base_path, "Core/Utility/Math")
	ALittle.Require(base_path, "Core/Utility/String")
	ALittle.Require(base_path, "Core/Utility/Time")
	ALittle.Require(base_path, "Core/Utility/Coroutine")
	ALittle.Require(base_path, "Core/Net/HttpFileReceiver")
	ALittle.Require(base_path, "Core/Net/HttpFileSender")
	ALittle.Require(base_path, "Core/Net/HttpReceiver")
	ALittle.Require(base_path, "Core/Net/HttpSender")
	ALittle.Require(base_path, "Core/Net/MsgCommon")
end

File: ALittle/Module/ALittleIDE/Other/GameLibrary/Core/JSScript/ALittle.js
{


let JSRequire = function(base_path, url, thread) {
	if (typeof(require) !== "undefined") {
		require("../../../" + base_path + url);
		thread(undefined);
		return;
	}
	let script = document.createElement("script");
	script.type = "text/javascript";
	script.async = "async";
	let error = thread.bind("load failed:" + url);
	script.onabort = error;
	script.onerror = error;
	script.ontimeout = error;
	script.onload = thread.bind(undefined);
	script.src = base_path + url + ".js";
	document.body.appendChild(script);
}

window.Require = function(base_path, url) {
	return new Promise(function(___COROUTINE, ___) {
		JSRequire(base_path, url, ___COROUTINE);
		return;
	});
}

window.RequireFromPaths = function(base_path, rel_path, file_list) {
	return new Promise(async function(___COROUTINE, ___) {
		let ___OBJECT_1 = file_list;
		for (let index = 1; index <= ___OBJECT_1.length; ++index) {
			let path = ___OBJECT_1[index - 1];
			if (path === undefined) break;
			path = path.substring(0, path.length - 8);
			await Require(base_path, rel_path + path);
		}
		___COROUTINE();
	});
}

window.RequireCore = function(base_path) {
	return new Promise(async function(___COROUTINE, ___) {
		await Require(base_path, "Core/JavaScript/JavaScriptClass");
		await Require(base_path, "Core/JavaScript/JavaScriptException");
		await Require(base_path, "Core/Reflect/ReflectRegister");
		await Require(base_path, "Core/Reflect/ReflectDefine");
		await Require(base_path, "Core/Utility/Log");
		await Require(base_path, "Core/Utility/List");
		await Require(base_path, "Core/Utility/Map");
		await Require(base_path, "Core/Utility/Math");
		await Require(base_path, "Core/Utility/String");
		await Require(base_path, "Core/Utility/Time");
		await Require(base_path, "Core/Utility/Coroutine");
		await Require(base_path, "Core/Net/HttpFileReceiver");
		await Require(base_path, "Core/Net/HttpFileSender");
		await Require(base_path, "Core/Net/HttpReceiver");
		await Require(base_path, "Core/Net/HttpSender");
		await Require(base_path, "Core/Net/MsgCommon");
		___COROUTINE();
	});
}

}